From arjan@infradead.org  Tue Sep 23 15:55:25 2008
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sat, 20 Sep 2008 19:09:00 -0700
Subject: debug: use dev_WARN() rather than WARN_ON() in device_pm_add()
To: Greg KH <greg@kroah.com>
Message-ID: <20080920190900.6d4ab35a@infradead.org>

From: Arjan van de Ven <arjan@linux.intel.com>

device_pm_add() has a WARN_ON that is showing relatively high on
kerneloops.org, but unfortunately the WARN_ON is less than useful
in that it doesn't print any information about what device is causing
the issue.

This patch fixes this by turning the WARN_ON() into the newly
introduces dev_WARN() which will print information about the
device in question.

Signed-off-by: Arjan van de Ven <arjan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/power/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -83,7 +83,7 @@ void device_pm_add(struct device *dev)
 		 * transition is in progress in order to avoid leaving them
 		 * unhandled down the road
 		 */
-		WARN_ON(true);
+		dev_WARN(dev, "Parentless device registered during a PM transaction\n");
 	}
 
 	list_add_tail(&dev->power.entry, &dpm_list);
